python/ray/data/_internal/execution/resource_manager.py

```diff
@@ -108,6 +121,8 @@ def get_global_limits(self) -> ExecutionResources:
             gpu=gpu,
             object_store_memory=object_store_memory,
         )
+        if self._op_resource_limiter is not None:
+            self._op_resource_limiter.on_global_limits_updated(self._global_limits)
```

Comment:
Felt confused at first about why we're calling `on_global_limits_updated` in `get_global_limits`. `get_and_update_global_limits` might be a clearer name, although maybe that's too verbose

Reply:
actually there is an issue, `on_global_limits_updated` may not get called before `update_resources`. 
I've changed the code to call `on_global_limits_updated` within the OpResourceLimiter itself.